### PR TITLE
Fix databricks resource setting

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/databricks/utils/databricks_utils.py [hunk 1]
```diff
@@ -85,3 +85,26 @@ def sanitize_labels(labels: Dict[str, str]) -> None:
         labels[key] = re.sub(r"[^0-9a-zA-Z-_\.]+", "_", value)[:63].strip(
             "-_."
         )
+
+
+def clean_requirements(requirements: List[str]) -> List[str]:
```
Comment (strickvl) [inline src/zenml/integrations/databricks/utils/databricks_utils.py:90]: This for sure wants a unit test

**Review by strickvl — CHANGES_REQUESTED** [review]

File: src/zenml/utils/package_utils.py [hunk 2]
```diff
@@ -48,3 +50,26 @@ def is_latest_zenml_version() -> bool:
...
+    Args:
+        requirements: List of requirements.
+
+    Returns:
+        Cleaned list of requirements
+    """
+    cleaned = {}
+    for req in requirements:
+        package = (
+            req.split(">=")[0]
```
Comment (schustmi) [inline src/zenml/utils/package_utils.py:67]: Are these pip requirements? In that case it could also be `~=` and `^=` I think?

Comment (strickvl) [reply src/zenml/utils/package_utils.py:67]: Yeah I think we need to handle these two use cases as well (the ~= (compatible release) and ^= (compatible release allowing pre-releases))

Comment (strickvl) [reply src/zenml/utils/package_utils.py:67]: Also I don't think we handle `package1>=1.0.0,<2.0.0` yet with this implementation. And there are a few other edge cases we're not handling

**Review by strickvl — CHANGES_REQUESTED** [review]

**Review by strickvl — APPROVED** [review]

**Pull request merged**